### DISCUSSION TITLE
useAnimatedGestureHandler and <GestureDetector> ?

### DISCUSSION THREAD

**Discussion by JS-GitRepo** [software-mansion/react-native-reanimated#4344] [Q&A]
The React Native Gesture Handler docs advise against the pattern listed in the Reanimated docs [here](https://docs.swmansion.com/react-native-reanimated/docs/api/hooks/useAnimatedGestureHandler), suggesting the use of "GestureDetector" instead of  "TapGestureHandler". The issue is that the required prop for GestureDetector, "gesture={}" does not accept useAnimatedGestureHandler() as an input. useAnimatedGestureHandler()'s type is infered as "OnGestureEvent-PanGestureHandlerGestureEvent-". In summary, is this hook now deprecated? 

The docs are very lacking when it comes to using GestureDetector with Reanimated. Any insight here? Thank you for your time.

**Comment by iway1** [comment]
Yeah we really, really need better docs on the new Gesture API. IDK if there are even complete docs for it. 

The new api is something like this:
```ts
  const gesture = useMemo(() => {
    return Gesture.Pan().onUpdate(event=>{/* do something */})
  }, []);
  
  return (
    <GestureDetector gesture={gesture} />
  )
```

The new API is way nicer to use. The hooks aren't necessary anymore

**Reply by JS-GitRepo** [reply]
Appreciate the reply! I was originally confused as to how to chain animations using this method, but they're actually really intuitive and easy to chain. Here's a codeblock that chains a few different actions within a gesture (not suggesting any of this is best practice btw, I'm fairly new to RN lol):
```js
        return {
			longPress: Gesture.LongPress()
				.onTouchesDown(() => {
					if (mediaNumber === 1 && m1Touch) {
						touchDownFunctionality();
						runOnJS(setM2Touch)(false);
					} else if (mediaNumber === 2 && m2Touch) {
						touchDownFunctionality();
						runOnJS(setM1Touch)(false);
					}
				})
				.onTouchesUp(() => {
					if (mediaNumber === 1 && m1Touch) {
						touchUpFunctionality();
						runOnJS(setM1Touch)(false);
						runOnJS(setM2Touch)(false);
						runOnJS(() => setTimeout(() => console.log("TEST"), 1000));
					} else if (mediaNumber === 2 && m2Touch) {
						touchUpFunctionality();
						runOnJS(setM1Touch)(false);
						runOnJS(setM2Touch)(false);
						// runOnJS(setTimeout)(() => console.log("TEST"), 1000);
						runOnJS(callGenerateMatchup)();
					}
				})
				.onTouchesCancelled(() => {
					if (!m1Touch && !m2Touch) {
					} else {
						touchCancelledFunctionality();
						runOnJS(setM1Touch)(true);
						runOnJS(setM2Touch)(true);
					}
				}),
		};
```
